src/balancerd/src/lib.rs

```diff
@@ -649,13 +649,33 @@ impl PgwireBalancer {
         envd_addr: SocketAddr,
         password: Option<String>,
         params: BTreeMap<String, String>,
-    ) -> Result<TcpStream, anyhow::Error>
+    ) -> Result<Conn<TcpStream>, anyhow::Error>
     where
         A: AsyncRead + AsyncWrite + AsyncReady + Send + Sync + Unpin,
     {
-        let mut mz_stream = TcpStream::connect(envd_addr).await?;
+        let mut tcp_stream = TcpStream::connect(envd_addr).await?;
         let mut buf = BytesMut::new();
 
+        FrontendStartupMessage::SslRequest.encode(&mut buf)?;
+        tcp_stream.write_all(&buf).await?;
```

Comment:
you need to call buf.clear() after writing, or else the startup message will be appended to the existing buffer and the ssl request message will be sent twice